src/panel/category/CategoryPanel.jsx

```diff
@@ -16,165 +16,136 @@ import { fire, listen, unListen } from 'src/libs/customEvents';
-    if (category && category !== prevProps?.poiFilters?.category) {
-      Telemetry.add(Telemetry.POI_CATEGORY_OPEN, { category });
-    }
-  }
+    const mapMoveHandler = listen('map_moveend', fetchData);
+    return () => { unListen(mapMoveHandler); };
+  }, [poiFilters, initialLoading]);
 
-  componentWillUnmount() {
-    unListen(this.mapMoveHandler);
-  }
+  useEffect(() => {
+    window.execOnMapLoaded(() => { fitMap(bbox); });
+  }, [bbox, poiFilters]);
 
-  fitMap() {
-    if (this.props.bbox) {
-      fire('fit_map', parseBboxString(this.props.bbox));
-      return;
-    }
+  useEffect(() => {
+    setInitialLoading(true);
+  }, [poiFilters]);
 
-    if (window.map.mb.isMoving && window.map.mb.isMoving()) {
-      /*
-        Do not trigger API search and zoom change when the map
-        is already moving, to avoid flickering.
-        The search will be triggered on moveend.
-      */
-      return;
+  useEffect(() => {
+    if (poiFilters.category) {
+      Telemetry.add(Telemetry.POI_CATEGORY_OPEN, { category: poiFilters.category });
     }
+  }, [poiFilters.category]);
 
-    // Apply correct zoom when opening a category
-    const currentZoom = window.map.mb.getZoom();
-
-    // Zoom < 5: focus on Paris
-    if (currentZoom < 5) {
-      window.map.mb.flyTo({ center: [2.35, 48.85], zoom: 12 });
-    } else if (currentZoom < 12) { // Zoom < 12: zoom up to zoom 12
-      window.map.mb.flyTo({ zoom: 12 });
-    } else if (currentZoom > 18) { // Zoom > 18: dezoom to zoom 18
-      window.map.mb.flyTo({ zoom: 18 });
-    } else {
-      // setting the same view still triggers the moveend event
-      window.map.mb.jumpTo({ zoom: currentZoom, center: window.map.mb.getCenter() });
-    }
-  }
-
-  fetchData = debounce(async () => {
-    const { category, query } = this.props.poiFilters;
-    const currentBounds = getVisibleBbox(window.map.mb, window.no_ui);
-
-    const extendBbox = this.state.initialLoading;
-    const { places, source, bbox: contentBbox, bbox_extended } = await IdunnPoi.poiCategoryLoad(
-      boundsToString(currentBounds),
-      MAX_PLACES,
-      category,
-      query,
-      extendBbox
-    );
-
-    this.setState({
-      pois: places,
-      dataSource: source,
-      initialLoading: false,
-    });
-
-    if (bbox_extended && contentBbox) {
-      // The returned bbox is sure to contain at least one POI.
-      // Extend the current one to include it.
-      fire('fit_map', currentBounds.extend(boundsFromFlatArray(contentBbox)), true);
-    }
-
-    fire('add_category_markers', places, this.props.poiFilters);
-    fire('save_location');
-  }, DEBOUNCE_WAIT, { leading: true })
-
-  selectPoi = poi => {
-    const { poiFilters } = this.props;
-    const { pois } = this.state;
+  const selectPoi = poi => {
     fire('click_category_poi', { poi, poiFilters, pois });
-  }
+  };
 
-  highlightPoiMarker = (poi, highlight) => {
+  const highlightPoiMarker = (poi, highlight) => {
```

Comment:
We can optionally move these functions which do not need to be inside the rendering function to the top. I approve this PR anyway, as it seems to works well !